ReportingServicesTools/Functions/Admin/Set-RsDatabase.ps1

```diff
@@ -112,22 +149,45 @@ function Set-RsDatabase
             $username = $rsWmiObject.WindowsServiceIdentityActual
             $password = ''
         }
-        
+
         else
         {
-            if ($DatabaseCredential -eq $null)
+            if ($DatabaseCredential -eq $nul)
             {
                 throw "No Database Credential specified! Database credential must be specified when configuring $DatabaseCredentialType authentication."
             }
             $username = $DatabaseCredential.UserName
             $password = $DatabaseCredential.GetNetworkCredential().Password
         }
         #endregion Validating authentication and normalizing credentials
-        
+
+        #region Validating admin authentication and normalizing credentials
+        $adminUsername = ''
+        $adminPassword = $null
+        # default is service account
+        if ($AdminDatabaseCredentialType -like 'serviceaccount' -or -not $AdminDatabaseCredential)
+        {
+            $adminUsername = $rsWmiObject.WindowsServiceIdentityActual
+            $adminPassword = ''
+        }
+
```

Comment:
I would remove from 168 to 172 and just have a condition for if $AdminDatabaseCredentialType is sql and then verify if $AdminDatabaseCredential is null

Reply:
Done. Pushing soon.